### PR TITLE
Support more syntax

### PR DESCRIPTION
**Pull request by Tatskaari**
Oops I also merged upstream in. I'll raise a separate PR which should make this easier to understand once landed (or you can just do that the normal way yourself)  

This adds support for syntax like this:

```
foo = (
    "foo"
    "bar"
    # test
    "baz"
)

bar = [
    "foo"
    "bar"
    # test
    "baz",
]

```

as well as removed `load()` so we treat it like a normal function call. It also handles `assert` (though we've getting rid of it soon) 

I can now parse all the build defs in the Please repo so I think we're there!

### CODE REVIEW

**Comment by peterebden** [comment]
Okay I've merged #1 , any chance you can rebase this? 
(I had a quick go but it seems to have way more changes than I expected still)

File: build/syntax.go [hunk 1]
```diff
@@ -273,6 +273,51 @@ func (x *StringExpr) Copy() Expr {
 	return &n
 }
 
+type MultiPartStringExpr struct {
```
Comment (Tatskaari) [inline build/syntax.go:276]: This isn't the most useful struct. We probably want some niceties like getting the decoded value our of it. Might be nice to iterate as we build out the SDK.

Comment (peterebden) [reply build/syntax.go:276]: Yeah or we just kill implicit string concatenation, it's not particularly nice anyway - I wasn't gonna have it initially but was having a big pain migrating.

**Comment by peterebden** [comment]
Right looks good. May be worth leaving a bunch of the load stuff in with just the top-level bits cut out in the parser to make merging upstream easier? But gonna go ahead with this for now.

**Comment by Tatskaari** [comment]
> Right looks good. May be worth leaving a bunch of the load stuff in with just the top-level bits cut out in the parser to make merging upstream easier? But gonna go ahead with this for now.

Yeah, good point. If we get rid of implicit string parsing, and assert, we'd be able to use the upstream one anyway. v17?

**Comment by peterebden** [comment]
I do quite like assert though :( Maybe we could do a Python 3 and make it a function!

**Pull request merged**